src/matrix/room/timeline/Timeline.js

```diff
@@ -234,10 +238,32 @@ export class Timeline {
     }
 
     /** @package */
-    addEntries(newEntries) {
+    async addEntries(newEntries) {
         this._addLocalRelationsToNewRemoteEntries(newEntries);
+        await this._loadRelatedEvents(newEntries);
         this._remoteEntries.setManySorted(newEntries);
     }
+
+    async _loadRelatedEvents(entries) {
+        const filteredEntries = entries.filter(e => !!e.relation);
+        for (const entry of filteredEntries) {
+            const id = entry.relatedEventId;
```

Comment:
Two things:
 - the id can be undefined/null
 - I don't think we should fetch the event for every relation. This would include for example the reacted-to-event for reactions (which are present at this stage in the timeline, but filtered out in TilesCollection later on). I would add a separate property (e.g. `contextEventId`, adding a comment on top of the getter to say what it is for) that only returns the event id in case of a reply.